XenAdmin/Wizards/RollingUpgradeWizard/RollingUpgradeWizardPrecheckPage.cs

```diff
@@ -197,45 +203,47 @@ public override string HelpID
                 }
             }
 
-            //iSL (StorageLink) check
-            checks.Add(new KeyValuePair<string, List<Check>>(Messages.CHECKING_STORAGELINK_STATUS, new List<Check>()));
-            checkGroup = checks[checks.Count - 1].Value;
-            foreach (Host host in SelectedServers)
+            //iSL (StorageLink) check - CA-223486: only for pre-Creedence
+            var preCreedenceServers = SelectedServers.Where(h => !Helpers.CreedenceOrGreater(h)).ToList();
+            if (preCreedenceServers.Any())
             {
-                checkGroup.Add(new HostHasUnsupportedStorageLinkSRCheck(host));
+                checks.Add(new KeyValuePair<string, List<Check>>(Messages.CHECKING_STORAGELINK_STATUS, new List<Check>()));
+                checkGroup = checks[checks.Count - 1].Value;
+                foreach (Host host in preCreedenceServers)
+                {
+                    checkGroup.Add(new HostHasUnsupportedStorageLinkSRCheck(host));
+                }
             }
 
             //Upgrading to Clearwater and above - license changes warning and deprecations
-            if(SelectedServers.Any(h=> !Helpers.ClearwaterOrGreater(h)))
+            var preClearwaterServers = SelectedServers.Where(h => !Helpers.ClearwaterOrGreater(h)).ToList();
+            if(preClearwaterServers.Any())
             {
-                //License changes
-                if (SelectedServers.Any(h => Host.GetEdition(h.edition) != Host.Edition.EnterpriseXD && 
-                                             Host.GetEdition(h.edition) != Host.Edition.XenDesktop))
                 {
                     checks.Add(new KeyValuePair<string, List<Check>>(Messages.CHECKING_LICENSING_STATUS, new List<Check>()));
                     checkGroup = checks[checks.Count - 1].Value;
-                    foreach (Host host in SelectedServers)
+                    foreach (Host host in preClearwaterServers)
                     {
-                        checkGroup.Add(new UpgradingFromTampaAndOlderCheck(host));
+                        if(HostNeedsLicenseCheck(host))

```

Comment:
Here we will end up with an empty group if none of the preClearwaterServers needs the license check.
I think it is OK to have a check for every server in preClearwaterServers, even if it would say OK for some of them (i.e. EnterpriseXD  and XenDesktop)